src/models/AssetPrice.ts

```diff
@@ -1,19 +1,23 @@
 import BigNumber from '../utils/BigNumber'
 
-export default class AssetRewards {
+export default class AssetPrice {
     public static readonly NETWORK_FEE_DENOMINATOR = BigNumber.from(10000)
 
     private totalPrice: BigNumber
 
     private rewards: Map<string, BigNumber>
 
+    private tokenAddress?: string
+
     public constructor()
 
     public constructor(_rewards: Map<string, BigNumber>)
 
     // eslint-disable-next-line no-dupe-class-members
     public constructor(address: string, amount: BigNumber)
 
+    public constructor(address: string, amount: BigNumber, tokenAddress: string)
+
     // eslint-disable-next-line no-dupe-class-members
     public constructor(..._params: any[]) {
```

Comment:
```suggestion
    public constructor()
    public constructor(_rewards: Map<string, BigNumber>)
    public constructor(address: string, amount: BigNumber)
    public constructor(address: string, amount: BigNumber, tokenAddress: string)
    public constructor(..._params: any[]) {
```

https://www.typescriptlang.org/docs/handbook/2/functions.html#overload-signatures-and-the-implementation-signature